### PR TITLE
ci: implement "console" stream for OS images to simplify debugging in customer environments

### PR DESCRIPTION
**Pull request by malt3**
### Proposed change(s)

- image: add AUTOLOGIN environment variable to conditionally enable serial console login
- ci: implement "console" stream for OS images

### Additional info

- image for testing: `ref/feat-ci-console-images/stream/console/v2.5.0-pre.0.20230113155604-88bdaa389b64`

### Checklist

- [x] Testing performed:
  - [x] [Image build pipeline](https://github.com/edgelesssys/constellation/actions/runs/3912850852)
  - [x] manual feature test
- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

File: .github/workflows/build-os-image.yml [hunk 1]
```diff
@@ -273,6 +274,7 @@ jobs:
           BOOTSTRAPPER_BINARY: ${{ github.workspace }}/build/bootstrapper
           DISK_MAPPER_BINARY: ${{ github.workspace }}/build/disk-mapper
           UPGRADE_AGENT_BINARY: ${{ github.workspace }}/build/upgrade-agent
+          AUTOLOGIN: ${{ inputs.stream == 'console' && 'true' || 'false' }}
```
Comment (Nirusu) [inline .github/workflows/build-os-image.yml:277]: Can we also do this for 'debug' and then remove the debugd code that does this otherwise?

Comment (malt3) [reply .github/workflows/build-os-image.yml:277]: Good idea! :+1:

File: debugd/internal/debugd/deploy/seriallogin.go [hunk 2]
```diff
@@ -1,41 +0,0 @@
...
-)
-
-// EnableAutoLogin installs a systemd unit override that allows passwordless root login
-// on the serial console.
-func EnableAutoLogin(ctx context.Context, fs afero.Fs, serviceManager serviceManager) error {
-	if err := fs.MkdirAll(path.Dir(debugd.GettyAutologinOverrideFilename), os.ModePerm); err != nil {
-		return fmt.Errorf("creating getty autologin override directory: %w", err)
-	}
-	if err := afero.WriteFile(fs, debugd.GettyAutologinOverrideFilename,
-		[]byte(debugd.GettyAutologinOverrideUnitContents), os.ModePerm); err != nil {
```
Comment (Nirusu) [inline debugd/internal/debugd/deploy/seriallogin.go:26]: You can likely also remove the two Getty constants in `constants.go` --> https://github.com/edgelesssys/constellation/blob/67f8336b9d2c3e935345559503053339fc1c25ad/debugd/internal/debugd/constants.go#L34

**Review by katexochen — APPROVED** [review]
Nice! :tada:

**Pull request merged**